woodwork/logical_types.py

```diff
@@ -30,6 +51,14 @@ def __eq__(self, other, deep=False):
     def __str__(self):
         return str(self.__class__)
 
+    @property
+    def primary_dtype(self):
+        return type(self).primary_dtype
```

Comment:
I'm not sure that it's immediately obvious why we need to do this (unless you're more familiar with python meta classes than I originally was!). Maybe it's just as simple as adding a couple of comments here or at the property definition on the metaclass indicating that this lets us include logic that gets executed whether or not it's being referenced from an instance object or just the class object itself.

When I was reading up on this stuff, I found this example really helpful:

```python
class MetaClass(type):
    @property
    def meta_prop(cls):
        return '(MetaClass.meta_prop) ' + cls.class_attr

class Class(object, metaclass=MetaClass):
    class_attr = ':D'
    
    @property
    def class_prop(self):
        return ('(Class.class_prop) ' + self.class_attr)
    
    @property
    def meta_prop(self):
        # For when we want to perform some operation beyond its definition at creation
        return ('(Class.meta_prop) ' + type(self).meta_prop)

# Access class attr
assert Class().class_attr == ':D'
assert Class.class_attr == ':D'

# Access class property
assert Class().class_prop == '(Class.class_prop) :D'
assert isinstance(Class.class_prop, property) # looks like '<property object at 0x16617f6d0>'

#  Access meta property
# This is the goal! Being able to reference a property on an uninstantiated class object
assert Class().meta_prop == '(Class.meta_prop) (MetaClass.meta_prop) :D'
assert Class.meta_prop == '(MetaClass.meta_prop) :D'
```

Reply:
@tamargrey Added a few comments to the property definitions. Do you think this is sufficient or is more needed?

Reply:
Yeah, I think they do a good job of clarifying what's going on here!